include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -768,23 +879,17 @@ namespace Utilities
 #    else
         if (n_procs > 99)
 #    endif
-          consensus_algo.reset(new NBX<T1, T2>(process, comm));
+          consensus_algo.reset(new NBX<T1, T2>());
         else
 #  endif
 #endif
           if (n_procs > 1)
-          consensus_algo.reset(new PEX<T1, T2>(process, comm));
+          consensus_algo.reset(new PEX<T1, T2>());
         else
-          consensus_algo.reset(new Serial<T1, T2>(process, comm));
-      }
-
-
+          consensus_algo.reset(new Serial<T1, T2>());
```

Comment:
In a side-by-side, you would see that this resetting of the member variable happened in the constructor. I had to move this into the `run()` function because we now have two constructors, only one of which knows how to choose the algorithm. That also means that the `consensus_algorithm` variable is actually only used in this one `run()` function any more. I could move it from a member variable to a local variable. I'll do so in a follow-up patch, but thought I'd try to keep things minimal for this PR.